From ffdf774b31c729fc6e9e4084806b3d59f106f65a Mon Sep 17 00:00:00 2001
From: anoopsjohn <anoopsamjohn@gmail.com>
Date: Fri, 10 Apr 2015 13:26:47 +0530
Subject: [PATCH 234/295] HBASE-13313 Skip the disabled table in mob
 compaction chore and MasterRpcServices.(Jingcheng)

- reverts to older style is enabled check

Reason: Bugfix
Author: Jingcheng Du
Ref: CDH-31077
---
 .../hadoop/hbase/master/MasterRpcServices.java     |    3 +++
 .../hbase/master/MobFileCompactionChore.java       |    3 +++
 2 files changed, 6 insertions(+)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java
index b60d56d..6a91b80 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java
@@ -1336,6 +1336,9 @@ public class MasterRpcServices extends RSRpcServices
    */
   private CompactRegionResponse compactMob(final CompactRegionRequest request,
     TableName tableName) throws IOException {
+    if (!master.getConnection().isTableEnabled(tableName)) {
+          throw new DoNotRetryIOException("Table " + tableName + " is not enabled");
+    }
     boolean isForceAllFiles = false;
     List<HColumnDescriptor> compactedColumns = new ArrayList<HColumnDescriptor>();
     HColumnDescriptor[] hcds = master.getTableDescriptors().get(tableName).getColumnFamilies();
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCompactionChore.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCompactionChore.java
index 53d3494..6f8d9a7 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCompactionChore.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MobFileCompactionChore.java
@@ -60,6 +60,9 @@ public class MobFileCompactionChore extends ScheduledChore {
       TableDescriptors htds = master.getTableDescriptors();
       Map<String, HTableDescriptor> map = htds.getAll();
       for (HTableDescriptor htd : map.values()) {
+        if (!master.getConnection().isTableEnabled(htd.getTableName())) {
+          continue;
+        }
         boolean reported = false;
         try {
           for (HColumnDescriptor hcd : htd.getColumnFamilies()) {
-- 
1.7.9.5

